### PR TITLE
Update MUI from v4 to v5

### PR DESCRIPTION
**Pull request by czmj**
This upgrade is done by following the [official document](https://mui.com/material-ui/migration/migration-v4) on the upgrade. This document is "getting started" on doing MUI upgrade, but we don't need to make (manually) any changes listed in other migration documents, because they are done by a migration tool from MUI called codemods.

In the migration guide, we can see that we meet the requirements for doing the upgrade, like minimal Typescript, React and Node.js version.

We can proceed with installation of new packages; `@mui/material` and `@mui/styles` are needed to replace old `@material-ui/core`, which is now removed.

> One of the biggest changes in v5 is the replacement of JSS for [Emotion](https://emotion.sh/docs/introduction) as a default styling solution.

However, we were using `styled-components` already and will continue doing so. Therefore, instead of installing `@emotion/react` and `@emotion/styled` peer dependencies, we install `@mui/styled-engine-sc`. We also need to [point Webpack](https://mui.com/material-ui/guides/styled-engine/#how-to-switch-to-styled-components) to use this package as the default styled engine.

Imports are changed and full list of changes can be found [here](https://mui.com/material-ui/migration/migration-v4/#replace-all-imports). But, this part is automatically migrated by MUI codemods:
```
npx @mui/codemod v5.0.0/preset-safe apps
npx @mui/codemod v5.0.0/preset-safe libs
```

Lastly, after upgrading to MUI v5, wrapping `ThemeProvider` from `@mui/material` inside another `ThemeProvider`, but from `styled-components` seems to swallow outer `theme` props, so we include all theme props inside MUI `ThemeProvider` and reduce the wrapping.

Seems like the rest of breaking changes aren't affecting us.

### CODE REVIEW

**Review by unocha-hpc — APPROVED** [review]
Checks have passed and this pull request is ready for manual review

File: package.json [hunk 1]
```diff
@@ -44,7 +44,8 @@
     "@babel/preset-env": "7.16.11",
     "@babel/preset-react": "7.16.7",
     "@babel/preset-typescript": "7.16.7",
-    "@material-ui/core": "^4.12.3",
+    "@emotion/react": "^11.9.0",
+    "@emotion/styled": "^11.8.1",
```
Comment (Pl217) [inline package.json]: In the [upgrade guide](https://mui.com/material-ui/migration/migration-v4/#introduction), inside introductory section, they say:

> One of the biggest changes in v5 is the replacement of JSS for [Emotion](https://emotion.sh/docs/introduction) as a default styling solution.

But, we are already using `styled-components`. In the [peer dependency section](https://mui.com/material-ui/migration/migration-v4/#peer-dependencies), they say you should use `@mui/styled-engine-sc` instead of Emotion packages.

I see you installed Emotion packages here (devDependency) and `@mui/styled-engine-sc` as a regular dependency.

File: package.json [hunk 2]
```diff
@@ -105,8 +106,13 @@
...
     "react-toastify": "8.2.0",
-    "styled-components": "5.3.5",
     "ts-jest": "27.1.4",
     "typescript": "4.6.3"
+  },
+  "dependencies": {
+    "@mui/material": "^5.8.1",
+    "@mui/styled-engine-sc": "^5.8.0",
+    "@mui/styles": "^5.8.0",
+    "styled-components": "^5.3.5"
```
Comment (Pl217) [inline package.json]: Why did you move the packages from `devDependencies` to regular dependencies? I don't know why, but all packages were marked as `devDependencies`. Maybe that's a Nx thing, but let's keep it for now.

File: libs/hpc-ui/src/lib/theme.tsx [hunk 3]
```diff
@@ -82,6 +82,18 @@ export const THEME = {
...
+      defaultProps: {
+        margin: 'normal',
+      },
+    },
+    MuiFormControl: {
+      defaultProps: {
+        margin: 'normal',
+      },
+    },
+  },
```
Comment (Pl217) [inline libs/hpc-ui/src/lib/theme.tsx:95]: Why are these changes made? When I run [codemods](https://mui.com/material-ui/migration/migration-v4/#run-codemods), these changes aren't made.

**Review by Pl217 — CHANGES_REQUESTED** [review]
My main complaint is that you haven't documented anything regarding this upgrade. I know this is extracted from #305, but we cannot have a single commit saying "Update MUI from v4 to v5" and call it a day.

I prefer git commits split between individual actions you took while doing the update. This may require you to skip commit checks (or make git bisecting harder later), but makes the upgrade process much easier to follow compared to a single big commit.

Pointing to the document used while doing the upgrade, commenting on specific parts, etc. all of that is lacking on this PR. I have gone through the update process myself and addressed my main complaints from code comments.

`styled-components` are used instead of Emotion, which means `@mui/styled-engine-sc` needed to be setup with Webpack and Typescript.
Also, I stumbled upon a problem where wrapping `ThemeProvider` from `@mui/material` inside another `ThemeProvider`, but from `styled-components` (this is all inside `libs/hpc-ui/src/lib/theme.tsx` file) seems to swallow outer `theme` props, so we need to include all theme props inside MUI `ThemeProvider` and reduce the wrapping.

This is all pushed as [pp/HPC-8326](https://github.com/UN-OCHA/hpc-cdm/tree/pp/HPC-8326) branch, so you can check it out. I will also update this PRs first comment with update process I used. Please review my commits and if you agree with the changes, I will move them over to this branch.

File: libs/hpc-ui/src/lib/theme.tsx (hunk 3 above)
Comment (czmj) [reply libs/hpc-ui/src/lib/theme.tsx:95]: Ok, thanks for your changes. Please merge them in to this branch.

The changes to margins are there because MUI has changed the default margin from "normal" to "none" on form components. CDM doesn't actually use any MUI form components so it's not a huge problem, but setting the default to what we will want for FTS Admin now made sense to me - I can move it to a separate PR if you'd like.

**Review by Pl217 — APPROVED** [review]

**Review by unocha-hpc — APPROVED** [review]
Checks have passed and this pull request is ready for manual review
